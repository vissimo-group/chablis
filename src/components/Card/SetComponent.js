import React from 'react';
import PropTypes from 'prop-types';
import Card from '.';

const SetComponent = ({
  accessibilityText,
  imageCard,
  imageLogo,
  imageCountry,
  width,
  height,
  darkenBackground,
}) => (
  <Card width={width} height={height} darkenBackground={darkenBackground}>
    {imageCard && (
      <img className="image-card" alt="Card" src={imageCard} />
    )}

    {imageCountry && (
      <img className="image-country" alt="Country" src={imageCountry} />
    )}

    {imageLogo && (
      <img className="image-logo" alt="Logo" src={imageLogo} />
    )}

    {accessibilityText && <span className="sr-only">{accessibilityText}</span>}
  </Card>
);

SetComponent.propTypes = {
  accessibilityText: PropTypes.string,
  imageLogo: PropTypes.string,
  imageCard: PropTypes.string,
  imageCountry: PropTypes.string,
  width: PropTypes.string,
  height: PropTypes.string,
  darkenBackground: PropTypes.bool,
};

SetComponent.defaultProps = {
  accessibilityText: '',
  darkenBackground: false,
  imageLogo: '',
  imageCard: '',
  imageCountry: '',
  width: '',
  height: '',
};

export default SetComponent;
