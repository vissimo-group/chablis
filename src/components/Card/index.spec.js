import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SetComponent from './SetComponent';

describe('Card', () => {

  it('Should render the default card', () => {
    const element = SetComponent({
      accessibilityText: 'Card Default',
      imageCard: 'bg-image',
      imageLogo: 'image-logo',
      imageCountry: 'image-country',
      width: '337px',
      height: '',
      darkenBackground: true,
    });

    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render the card without flag', () => {
    const element = SetComponent({
      accessibilityText: 'Card Default',
      imageCard: 'bg-image',
      imageLogo: 'image-logo',
      imageCountry: '',
      width: '337px',
      height: '',
      darkenBackground: true,
    });

    const component = mount(element);
    expect(component.find('.image-country')).toHaveLength(0);
  });

  it('Should render a fluid component', () => {
    const element = SetComponent({
      accessibilityText: 'Card Default',
      imageCard: 'bg-image',
      imageLogo: 'image-logo',
      imageCountry: 'image-country',
      width: '',
      height: '',
      darkenBackground: true,
    });

    const component = mount(element);
    expect(component.props().width).toEqual('');
  });

  it('Should render an empty card', () => {
    const element = SetComponent({
      accessibilityText: '',
      imageCard: '',
      imageLogo: '',
      imageCountry: '',
      width: '',
      height: '',
      darkenBackground: true,
    });

    const component = mount(element);

    expect(component.props().width).toEqual('');
    expect(component.props().height).toEqual('');

    expect(component.find('.image-country')).toHaveLength(0);
    expect(component.find('.image-logo')).toHaveLength(0);
    expect(component.find('.sr-only')).toHaveLength(0);
  });
});
