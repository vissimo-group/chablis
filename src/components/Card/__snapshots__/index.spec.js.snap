// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render the default card 1`] = `
<div
  className="sc-bdVaJa buZcqG"
  height=""
  width="337px"
>
  <img
    alt="Card"
    className="image-card"
    src="bg-image"
  />
  <img
    alt="Country"
    className="image-country"
    src="image-country"
  />
  <img
    alt="Logo"
    className="image-logo"
    src="image-logo"
  />
  <span
    className="sr-only"
  >
    Card Default
  </span>
</div>
`;
